plan_test.go

```diff
@@ -844,11 +818,19 @@ func TestQueryPlanNoUnnessecaryID(t *testing.T) {
 		  {
 			"ServiceURL": "A",
 			"ParentType": "Query",
-			"SelectionSet": "{ movies { title } }",
+			"SelectionSet": "{ movies { title _bramble_id: id } }",
```

Comment:
I believe this test here represented another broken scenario: the boundary type (Movie) was being requested without an `id`. Digging into the planning flow, the omission was caused by a nuance in the `childstep` tracking. I've eliminated the `childstep` logic entirely as its sole purpose appears to have been controlling selection redundancies.